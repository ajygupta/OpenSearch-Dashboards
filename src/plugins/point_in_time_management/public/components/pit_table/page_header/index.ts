/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export { PageHeader } from './page_header';
