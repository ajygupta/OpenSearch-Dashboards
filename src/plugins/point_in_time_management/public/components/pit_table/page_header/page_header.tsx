/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  EuiButton,
  EuiPageContentHeader,
  EuiPageContentHeaderSection,
  EuiText,
  EuiTitle,
} from '@elastic/eui';
import { FormattedMessage } from '@osd/i18n/react';
import React from 'react';

export const PageHeader = () => (
  <>
    <EuiPageContentHeader>
      <EuiPageContentHeaderSection>
        <EuiTitle>
          <h1>
            <FormattedMessage id="pitManagement.header.pitTitle" defaultMessage="Point in Time" />
          </h1>
        </EuiTitle>
      </EuiPageContentHeaderSection>
      <EuiButton fill={true} iconType="plusInCircle" data-test-subj="createPITBtnInHeader">
        <FormattedMessage id="pitManagement.header.createPitButton" defaultMessage="Create PIT" />
      </EuiButton>
    </EuiPageContentHeader>
    <EuiText size="s">
      <p>
        <FormattedMessage
          id="pitManagement.pitDescription"
          defaultMessage="Create and manage point in time objects."
        />
      </p>
    </EuiText>
  </>
);
