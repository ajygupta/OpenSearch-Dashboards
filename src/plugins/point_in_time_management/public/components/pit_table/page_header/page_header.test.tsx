/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { PageHeader } from './page_header';
import { shallow } from 'enzyme';

describe('PageHeader', () => {
  it('should render normally', () => {
    const component = shallow(<PageHeader />);

    expect(component).toMatchSnapshot();
  });
});
