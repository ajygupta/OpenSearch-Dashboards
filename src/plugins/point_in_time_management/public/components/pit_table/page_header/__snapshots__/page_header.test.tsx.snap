// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader should render normally 1`] = `
<Fragment>
  <EuiPageContentHeader>
    <EuiPageContentHeaderSection>
      <EuiTitle>
        <h1>
          <FormattedMessage
            defaultMessage="Point in Time"
            id="pitManagement.header.pitTitle"
            values={Object {}}
          />
        </h1>
      </EuiTitle>
    </EuiPageContentHeaderSection>
    <EuiButton
      data-test-subj="createPITBtnInHeader"
      fill={true}
      iconType="plusInCircle"
    >
      <FormattedMessage
        defaultMessage="Create PIT"
        id="pitManagement.header.createPitButton"
        values={Object {}}
      />
    </EuiButton>
  </EuiPageContentHeader>
  <EuiText
    size="s"
  >
    <p>
      <FormattedMessage
        defaultMessage="Create and manage point in time objects."
        id="pitManagement.pitDescription"
        values={Object {}}
      />
    </p>
  </EuiText>
</Fragment>
`;
